src/chainstate/burn/operations/peg_out_fulfill.rs

```diff
@@ -0,0 +1,265 @@
+use stacks_common::codec::StacksMessageCodec;
+use stacks_common::types::chainstate::StacksBlockId;
+
+use crate::burnchains::BurnchainBlockHeader;
+use crate::burnchains::BurnchainTransaction;
+use crate::chainstate::burn::Opcodes;
+use crate::types::chainstate::StacksAddress;
+use crate::types::Address;
+
+use crate::chainstate::burn::operations::Error as OpError;
+use crate::chainstate::burn::operations::PegOutFulfillOp;
+
+impl PegOutFulfillOp {
+    pub fn from_tx(
+        block_header: &BurnchainBlockHeader,
+        tx: &BurnchainTransaction,
+    ) -> Result<Self, OpError> {
+        if tx.opcode() != Opcodes::PegOutFulfill as u8 {
+            warn!("Invalid tx: invalid opcode {}", tx.opcode());
+            return Err(OpError::InvalidInput);
+        }
+
+        let (amount, recipient) = if let Some(Some(recipient)) = tx.get_recipients().first() {
+            (recipient.amount, recipient.address.clone())
+        } else {
+            warn!("Invalid tx: Output 2 not provided");
+            return Err(OpError::InvalidInput);
+        };
+
```

Comment:
You're also going to need to look for output 3, which is a payment to the peg wallet.

Reply:
I've added output 3 to the request op. There's no output 3 in the fulfill op afaik. However, I'll extend the data of this op to contain a pointer to the peg out request op.